controllers/front/PasswordController.php

```diff
@@ -121,16 +121,29 @@ protected function changePassword()
                 $this->errors[] = $this->trans('You cannot regenerate the password for this account.', array(), 'Shop.Notifications.Error');
             }
 
-            // Case if both password params not posted or different, then "change password" form is not POSTED, show it.
-            if (!(Tools::isSubmit('passwd'))
-                || !(Tools::isSubmit('confirmation'))
-                || ($passwd = Tools::getValue('passwd')) !== ($confirmation = Tools::getValue('confirmation'))
-                || !Validate::isPasswd($passwd) || !Validate::isPasswd($confirmation)) {
-                // Check if passwords are here anyway, BUT does not match the password validation format
-                if (Tools::isSubmit('passwd') || Tools::isSubmit('confirmation')) {
-                    $this->errors[] = $this->trans('The password and its confirmation do not match.', array(), 'Shop.Notifications.Error');
+            if ($isSubmit = Tools::isSubmit('passwd')) {
+                // If password is submitted validate pass and confirmation
+                if (!$passwd = Tools::getValue('passwd')) {
+                    $this->errors[] = $this->trans("The password is missing: please enter your new password.", array(), 'Shop.Notifications.Error');
```

Comment:
It is already localized in _Admin.Login.Notification_. Can a wording for the FO can be localized in the BO files? If yes, let it be attached to the _Admin.Login.Notification_ file. If no, I'm okay with this suggestion of localizing it in _Shop.Notifications.Error_. :-)

Reply:
The very best way should be abstract the password recover management at higher level: it's the same function in FO and in BO.
If you look at the chunk, it was written with some redundancy  3 years ago

> [*] Security : Add "change password" screen accessible from a "forgot password" query sent by mail. Part of #PSCSX-5770 for Admin only.  (cherry picked from commit 56133c5b34ab88e1e99c3a6fdce27b065be00d1e)

It's my first contribution on this project so I don't know what is the policy on small fix / feature redesign.

I think the unification of FO / BO password recover should be a feature for a next release

Reply:
@eternoendless, what's your opinion about it?

Reply:
What happens now?

Reply:
> Can a wording for the FO can be localized in the BO files?

I don't think we should mix translations from FO and BO, that's part of the point of having translation domains, isn't it?

> I think the unification of FO / BO password recover should be a feature for a next release

I'm in favor of unifying some of the logic later on, but we should first list the functional differences (if any) between the two processes. Let's also keep in mind that wording domains should probably kept separate, as I said before.

Reply:
Right, thanks @eternoendless. @web-fu, let's keep the _Shop.Notifications.Error_ domain then!